src/app/cli/src/commands/pull_command.rs

```diff
@@ -104,11 +107,14 @@ impl PullCommand {
     async fn pull_multi(
         &self,
         listener: Option<Arc<dyn PullMultiListener>>,
+        current_account_name: Option<AccountName>,
     ) -> Result<Vec<PullResponse>, CLIError> {
         let dataset_refs: Vec<_> = filter_datasets_by_any_pattern(
             self.dataset_repo.as_ref(),
             self.search_svc.clone(),
             self.refs.clone(),
+            current_account_name,
+            self.all,
```

Comment:
I don't think that switch `-all` and list of references are a valid combination. If I use `-all`, I would expect to update all of my datasets without specifying any patterns/refs. If that is so, then it's quite weird to mix extraction of all account datasets and "-all" switch in 1 filtering algorithm.

Reply:
We will not allow the use of both refs and `--all`. But as we are encapsulating a logic of fetching datasets to process, I decided it's a good place to handle this flag. And one more reason that `--all` will not mean like really all records in the case of multitenant workspace.

Reply:
Agreed to bring changes and move `-all` flag handling back to pull implementation level